### PR TITLE
Fix-up Post Editor’s preferences modal

### PR DESCRIPTION
**Pull request by stokesman**
Follow-up to #35142. This is meant to do a few things:
- Fix clipping of focus styles and breakage of the sticky positioning of the category titles in the Blocks screen (since #35332).
- Refine visuals of the preferences modal (at < medium breakpoints) closer to what we had before #35142.
- Clean up some outdated CSS for customizing the legacy navigation component and its children.

It's worth noting the approach for the second point on improving visuals is by edits to only component structure and props as opposed to adding custom CSS. The idea is to follow the approach Riad proposed:

> Ideally we shouldn't have to customize things other than with using the regular props offered by the UI components.

https://github.com/WordPress/gutenberg/pull/35142#issuecomment-929020861

While the visuals can be further improved, some of it may be better done with careful improvement to component styles.

## How has this been tested?
Manually.

### Before (since #35332) clipping and sticky breakage are evident

https://user-images.githubusercontent.com/9000376/136718890-1fc0555b-dda3-4347-ba57-79b265c19b20.mp4

### With this PR clipping and sticky issues resolved

https://user-images.githubusercontent.com/9000376/136718923-62af3e91-31a2-4e8d-a9a6-43ab0acc0db2.mp4

### Blocks screen
| `Before` | `After…` |
| --- | --- |
| before-edit-post-prefs-blocks | after-edit-post-prefs-blocks |

### Root menu
| `Before` | `After…` |
| --- | --- |
| before-edit-post-prefs | after-edit-post-prefs |

## Types of changes
Enhancement

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/edit-post/src/components/preferences-modal/index.js
Comment (stokesman) [inline packages/edit-post/src/components/preferences-modal/index.js:333]: Putting the `Card`s and `CardBody`s inside the `NavigatorScreen`s prevents the clipping of focus indicators and allows the root screen to use a padding different from the child screens.

File: packages/edit-post/src/components/preferences-modal/style.scss
Comment (stokesman) [inline packages/edit-post/src/components/preferences-modal/style.scss]: The comment refers to something like #35518 being merged

Comment (ciampo) [reply packages/edit-post/src/components/preferences-modal/style.scss]: Yeah, it's probably a good idea to merge #35518 first and later rebase this PR on top.

**Review by ciampo — COMMENTED** [review]
Hey @stokesman , thank you again for noticing the regression first and working on this PR!

I personally like the direction taken here — especially about removing custom CSS in favour of components composition.

I will also add other folks who know the preferences modal design and functionality a bit better to help with the review. (@jasmussen @ntsekouras and @jorgefilipecosta )

**Comment by jasmussen** [comment]
This is beautiful work. Notably I love how much bespoke CSS just for this one component has been _removed_, in favor of using the components to lay things out. Any layout issues present can then be fixed in the components themselves, ideally benefitting the whole system.

This looks good at the wide breakpoint

Screenshot 2021-10-12 at 10 37 05

It also looks pretty good at the small breakpoint, which I understand was the focus of this one:

Screenshot 2021-10-12 at 10 37 20

However this one feels like a small step backwards:
Screenshot 2021-10-12 at 10 37 20

It's nice that the focus style is not clipped, but the button is fairly huge for what it needs to be. Can it be inline-block?

The subheading, "Blocks", also feels like it sits alone, and with its normal font size and weight it looks like text rather than a heading. Can it be a big heading like it was before? 

An option would also be to have an icon-only back button, and then have the "Blocks" subsection item sit immediately after it, horizontally.

**Comment by stokesman** [comment]
Thanks for the review @jasmussen! I've pushed some updates—the main one to try your suggestion:

> An option would also be to have an icon-only back button, and then have the "Blocks" subsection item sit immediately after it, horizontally.

post-preferences-navigator-card-header-text16-wcolor

I didn't want to go back to the heading size as it was close to but slightly off from the modal header font size. I've made it the same size as the modal header, but kept it a lighter weight. Though it's not visually apparent, I also got rid of the actual `h2` that was used because the headings that follow are h2s (and that is appropriate for the screen @>medium breakpoints where the screen heading does not exist). The other minor point is I removed the top margin from the the inner section headings to tighten-up the gap under the screen/card header.

I didn't update the snapshot this time around in case we need to make more changes. That's a test that will fail.

File: packages/edit-post/src/components/preferences-modal/style.scss
Comment (stokesman) [reply packages/edit-post/src/components/preferences-modal/style.scss]: Rebased and these lines are gone 🎉

File: packages/edit-post/src/components/preferences-modal/index.js
Comment (jasmussen) [inline packages/edit-post/src/components/preferences-modal/index.js]: This color appears to come from some React Native files that we should not be using in most cases, and frankly should consider deprecating in the native pieces.

**Review by jasmussen — APPROVED** [review]
This is what I see:

modal

This is a step forward in a few ways:

- It replaces a ton of bespoke CSS in favor of using the new components. This is a massive gain, well done.
- The icon-only back works well
- It's a good small-viewport experience

Because of the code quality improvements, it does not preclude future and further enhancements.

It also does surface some challenges, though:

- There's still a `font-size: 0.9rem;` rule in the code, which feels arbitrary and ill-advised. It also makes the back label 16px font size, which feels too big. But since this is existing code that you have not touched, the issue should not block this PR.

In that vein, it feels like enough of a step forward that following up on the font size is something that we could look at separately as we continue the extraction of bespoke CSS, which really is a noble goal. 

_I did leave one small note about a color that we shouldn't be using, and which can be removed before this PR lands._

Thank you.

File: packages/edit-post/src/components/preferences-modal/index.js
Comment (ciampo) [reply packages/edit-post/src/components/preferences-modal/index.js]: Since this originally was a `h2`, ideally we should use [`Heading`](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/heading), but I'm afraid that it wouldn't support the styles required here (one more for https://github.com/WordPress/gutenberg/issues/35464)

As an alternative, we could for now simply add `as="h2"` to the `Text` component 

Also, I wonder if there's a better alternative to adding a hardcoded hex color inline? Maybe we should create a classname instead and add that styling to `styles.scss` (especially if that color already exists as a scss variable)

**Review by ciampo — COMMENTED** [review]
> There's still a font-size: 0.9rem; rule in the code, which feels arbitrary and ill-advised. It also makes the back label 16px font size, which feels too big. But since this is existing code that you have not touched, the issue should not block this PR.

Yeah, definitely. Once components like `Text` and `Heading` feel a bit more mature, it'd be great also to re-write parts of the UI with it — and the preferences modal feels like a good self-contained piece of UI.

File: packages/edit-post/src/components/preferences-modal/index.js
Comment (stokesman) [reply packages/edit-post/src/components/preferences-modal/index.js]: The color shall be gone. It was only meant to be temporary until further updates bring color consistency. We can let the inconsistency be a spur for such improvements (if anyone is sensitive enough 😄 ).

> Since this originally was a h2

An `h2` could make sense here (in my first commit I had the `<Text as="h2"…`) but I realized all the headings that follow are also `h2`. If we simply make those `h3` it's not correct at greater than medium breakpoints where the screen heading doesn't exist. Instead of adding any complexity to change heading level by breakpoint I decided to not use the `h2`. It seems better though it's definitely not the focus of the PR.

**Comment by stokesman** [comment]
Thanks for reviewing again @jasmussen and @ciampo 🙇 

> It replaces a ton of bespoke CSS in favor of using the new components. This is a massive gain, well done.

Credit really goes to Riad for putting in the new component (while leaving the CSS 😄 ).

> as we continue the extraction of bespoke CSS, which really is a noble goal.

And there is indeed more of that to be done for the preferences modal. I am eager to land the fixes in this PR without getting into more of that for now. I might have exercised more restraint in these changes just to get the sure-wins in sooner. Thanks for taking it in stride Joen and Marco!

**Pull request merged**
